### ISSUE TITLE
resource ibm_cos_bucket deleted/created when no changes to the configuration

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#816]
### Terraform Version
```
~/tmp/tfbugs/tfcosbucket $ tf version
Terraform v0.11.14
+ provider.ibm v0.18.0

Your version of Terraform is out of date! The latest version
is 0.12.12. You can update by downloading from www.terraform.io/downloads.html
```

### Affected Resource(s)
```
variable region {
  default = "us-south"
}
variable cos_instance_name {
  default = "bug00"
}
variable image_import_bucket {
  default = "image-import"
}
variable prefix {
  default = "bug00"
}

locals {
  cos_bucket_region = "${var.region}"
  cos_bucket_name = "${format("%s-%s", var.prefix, var.image_import_bucket)}"
}

variable ibmcloud_api_key {}

provider "ibm" {
  region             = "${var.region}"
  ibmcloud_api_key = "${var.ibmcloud_api_key}"
}

resource "ibm_resource_instance" "cos_instance" {
  name              = "${var.cos_instance_name}"
  service           = "cloud-object-storage"
  plan              = "standard"
  location          = "global"
}

resource "ibm_cos_bucket" "bucket" {
  bucket_name = "${local.cos_bucket_name}"
  resource_instance_id = "${ibm_resource_instance.cos_instance.id}"
  region_location = "${local.cos_bucket_region}"
  storage_class = "standard"
}

output cos_instance_name { value = "${var.cos_instance_name}" }
output cos_instance_id { value = "${ibm_resource_instance.cos_instance.id}" }
output cos_bucket_region { value = "${local.cos_bucket_region}" }
output cos_bucket_name { value = "${local.cos_bucket_name}" }

```

Run apply two times, notice the second time results in the deletion and add of the bucket.

```
~/tmp/tfbugs/tfcosbucket $ tfa
ibm_resource_instance.cos_instance: Creating...
  crn:      "" => "<computed>"
  location: "" => "global"
  name:     "" => "bug00"
  plan:     "" => "standard"
  service:  "" => "cloud-object-storage"
  status:   "" => "<computed>"
ibm_resource_instance.cos_instance: Still creating... (10s elapsed)
ibm_resource_instance.cos_instance: Creation complete after 19s (ID: crn:v1:bluemix:public:cloud-object-stor...2e0484b0-95ce-46bd-8638-40b2665154e8::)
ibm_cos_bucket.bucket: Creating...
  bucket_name:          "" => "bug00-image-import"
  crn:                  "" => "<computed>"
  region_location:      "" => "us-south"
  resource_instance_id: "" => "crn:v1:bluemix:public:cloud-object-storage:global:a/713c783d9a507a53135fe6793c37cc74:2e0484b0-95ce-46bd-8638-40b2665154e8::"
  s3_endpoint_private:  "" => "<computed>"
  s3_endpoint_public:   "" => "<computed>"
  storage_class:        "" => "standard"
ibm_cos_bucket.bucket: Creation complete after 2s (ID: crn:v1:bluemix:public:cloud-object-stor...et:bug00-image-import:meta:rl:us-south)

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

cos_bucket_name = bug00-image-import
cos_bucket_region = us-south
cos_instance_id = crn:v1:bluemix:public:cloud-object-storage:global:a/713c783d9a507a53135fe6793c37cc74:2e0484b0-95ce-46bd-8638-40b2665154e8::
cos_instance_name = bug00
~/tmp/tfbugs/tfcosbucket $ tfa
ibm_resource_instance.cos_instance: Refreshing state... (ID: crn:v1:bluemix:public:cloud-object-stor...2e0484b0-95ce-46bd-8638-40b2665154e8::)
ibm_cos_bucket.bucket: Refreshing state... (ID: crn:v1:bluemix:public:cloud-object-stor...et:bug00-image-import:meta:rl:us-south)
ibm_cos_bucket.bucket: Destroying... (ID: crn:v1:bluemix:public:cloud-object-stor...et:bug00-image-import:meta:rl:us-south)
ibm_cos_bucket.bucket: Destruction complete after 1s
ibm_cos_bucket.bucket: Creating...
  bucket_name:          "" => "bug00-image-import"
  crn:                  "" => "<computed>"
  region_location:      "" => "us-south"
  resource_instance_id: "" => "crn:v1:bluemix:public:cloud-object-storage:global:a/713c783d9a507a53135fe6793c37cc74:2e0484b0-95ce-46bd-8638-40b2665154e8::"
  s3_endpoint_private:  "" => "<computed>"
  s3_endpoint_public:   "" => "<computed>"
  storage_class:        "" => "standard"

Error: Error applying plan:

1 error occurred:
	* ibm_cos_bucket.bucket: 1 error occurred:
	* ibm_cos_bucket.bucket: BucketAlreadyExists: Container bug00-image-import exists
	status code: 409, request id: 4dd0db67-9383-4e70-a3b8-de9e96cb9421, host id:
```

**Comment by powellquiring** [comment]
@n-marton

**Comment by n-marton** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/823 this pr already merged with the fix

**Comment by n-marton** [comment]
fix has been added in 0.19.0, this can be closed
